gcp_variant_transforms/options/variant_transform_options.py

```diff
@@ -208,9 +212,31 @@ def add_arguments(self, parser):
               'core machine using two processes should help interleaving I/O '
               'vs CPU bound work.'))
 
-    # TODO(bashir2): Add validate() to check --vep_* arguments are sound when
-    # --run_annotation_pipeline is set (for example --annotation_output_dir
-    # should start with gs://).
+  def validate(self, parsed_args):
+    # type: (argparse.Namespace) -> None
+    args_dict = vars(parsed_args)
+    if args_dict[AnnotationOptions._RUN_FLAG]:
```

Comment:
You should be able to just reference `parsed_args.run_annotation_pipeline` natively here without creating the constants. See `BigQueryWriteOptions` validate method for an example.

Reply:
Yes I know but I actually preferred this because I already have to declare these constants. The reason is that the static typing system has no idea about these flags (unlike say a named tuple) so for example if I have a typo in the name of the flag it is not different from the correct spelling (both are errors from the type checking perspective).

If you feel strongly about this, I can change it.

Reply:
I know the common pattern for argparse and have used it in many places, including access to these very flags _outside_ this module (check vep_runner for example). Here though, as I said, I have to define the constants anyways. So forget about the type checking; are you suggesting that I drop the AnnotationOptions constants (e.g., _RUN_FLAG, _VEP_*, etc.)? Please take a look at this whole function not just this particular line. I think the answer is "no", because I have to repeat those flag texts several times including inside raw strings (typos there are not going to be caught by anything, including unit-tests). Now if we agree to keep the constants, then I think it makes the code more understandable to use those constants everywhere in this module, including flag declarations and access to them.

If on the other hand, you feel we need to drop the constants too, then sure I repeat the strings everywhere (but I don't think that is a good choice).

Reply:
ok, sounds good. let's keep it in this case (I just don't want it to be a common pattern for all of our args, but seems like it would be isolated to just annotations for the reasons you mentioned).